Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -697,20 +695,17 @@ void SourceBufferPrivate::append(Ref<SharedBuffer>&& buffer)
         // Resolve the changes in TrackBuffers' buffered ranges
         // into the SourceBuffer's buffered ranges
         auto trackBuffers = trackBuffersRanges();
-        if (isAttached())
-            m_client->sourceBufferPrivateTrackBuffersChanged(trackBuffers);
+        m_client->sourceBufferPrivateTrackBuffersChanged(trackBuffers);
 
         Vector<Ref<GenericPromise>> promises;
         promises.append(updateBufferedFromTrackBuffers(trackBuffers));
-        if (m_client) {
-            if (m_groupEndTimestamp > duration()) {
-                // https://w3c.github.io/media-source/#sourcebuffer-coded-frame-processing
-                // 5. If the media segment contains data beyond the current duration, then run the duration change algorithm with new
-                // duration set to the maximum of the current duration and the group end timestamp.
-                promises.append(m_client->sourceBufferPrivateDurationChanged(m_groupEndTimestamp));
-            }
-            m_client->sourceBufferPrivateReportExtraMemoryCost(totalTrackBufferSizeInBytes());
+        if (m_groupEndTimestamp > duration()) {
+            // https://w3c.github.io/media-source/#sourcebuffer-coded-frame-processing
+            // 5. If the media segment contains data beyond the current duration, then run the duration change algorithm with new
+            // duration set to the maximum of the current duration and the group end timestamp.
+            promises.append(m_client->sourceBufferPrivateDurationChanged(m_groupEndTimestamp));
         }
+        m_client->sourceBufferPrivateReportExtraMemoryCost(totalTrackBufferSizeInBytes());
```

Comment:
Ditto.

Reply:
ok, will change them all to use `client()` ; we need a private private class member :)